### ISSUE TITLE
oneOf or has not working

### ISSUE THREAD

**Issue by vikram1992** [mekanika/skematic#39]
These two rules are not working only first time. Here is the attached screenshot.

This screenshot contains validators and rules.
<img width="1280" alt="screen shot 2018-05-16 at 11 03 23" src="https://user-images.githubusercontent.com/11718385/40098379-cdad4b6c-58f8-11e8-8baf-1a58c542c5cd.png">

This screenshot contains validations thrown by the library.
<img width="1280" alt="screen shot 2018-05-16 at 11 03 31" src="https://user-images.githubusercontent.com/11718385/40098381-cdddf668-58f8-11e8-8a46-1727a5d9786d.png">

**Comment by cayuu** [comment]
Can you confirm which version of Skematic you're using?

**Comment by vikram1992** [comment]
Its 2.2.3

**Comment by cayuu** [comment]
The latest version is 3.5.1. Have you tried that to see if it fixes your issue?

**Comment by vikram1992** [comment]
Yeah it fixes, need to change the whole base structure. And also sub model validation is supported but if parent key of sub-model is not there, even though the validation passes in new version. Can you help me in this.

**Comment by cayuu** [comment]
Can you provide an example of what you mean?

**Comment by vikram1992** [comment]
Okay I have handled it through helper class. But the example is like below

inventoryList: {
            model: {
                productId: {
                    type: Skematic.STRING,
                    required: true,
                    errors: {
                        required: errors.COMMON.REQUIRED.FIELD,
                    }
                },
                variantId: {
                    type: Skematic.STRING,
                    required: true,
                    errors: {
                        required: errors.COMMON.REQUIRED.FIELD,
                    }
                },
                inventory: {
                    type: Skematic.INTEGER,
                    required: true,
                    errors: {
                        required: errors.COMMON.REQUIRED.FIELD,
                    }
                },
            }


If the key inventory list is empty or undefined, then validation is still passed which should be failed in my case.
